# Turtle-Egg-Microbiome
Data and coding used for Master's thesis on loggerhead and green sea turtle egg microbiomes

**What's Included in Repository**
1. R Code used for Microbiome Analysis and Sand Grain Size and Sorting Coefficient determination
2. Excel file with list of _POTENTIAL_ egg pathogens
3. Krona Plot HTMLs
4. Full SIMPER analysis lists

**Project summary**

The primary aims of this study characterized and compared the microbial communities (microbiomes) from the i) cloaca of nesting sea turtles; ii) sand within and surrounding the nests and iii) hatched and unhatched eggshells from two important species, the loggerhead (_Caretta caretta_) and green (_Chelonia mydas_) turtles. Samples were collected from nests in Fort Lauderdale and Hillsboro beaches in Broward County, Florida, where local nesting consisted of approximately 95% loggerhead, 5% green and 1% leatherback turtles (_Dermochelys coriacea_). Illumina MiSeq high throughput sequencing of 16S rRNA V3-V4 region amplicons was performed on 27 cloaca, 54 sand, and 162 egg samples. This study provides new insights into the microbial communities associated with egg failure in sea turtle nests and unveiled information about the impacts the environment has on microbial introduction.

**Total Sample Breakdown**

27 Nests
-20 Loggerhead
  -11 from Fort Lauderdale
  -9 from Hillsboro
-7 Green Turtles
  -2 from Fort Lauderdale
  -5 from Hillsboro

_Samples Per Nest_
*1 cloaca swab
*3 hatched egg swabs
*3 unhatched egg swabs
*1 nest sand sample
*1 control sand sample

All sampling approved and conducted by authorized personnel under Florida Fish and Wildlife Commision (FWC) Marine Turtle Permits (MTP) MTP-21-255 and MTP-21-214

Master's thesis can be found at 

Publication version can be accessed at

**Please cite this paper if you use this repository:**
